src/rust-util/pelikan/src/core/mod.rs

```diff
@@ -0,0 +1,129 @@
+//! Wrapper for the pelikan `core` module.
+
+pub mod admin;
+
+use ccommon::buf::OwnedBuf;
+use ccommon_sys::buf;
+
+use pelikan_sys::core::data_processor;
+
+use libc::c_int;
+use std::ffi::c_void;
+use std::mem::MaybeUninit;
+
+#[repr(C)]
+pub enum DataProcessorError {
+    Error = -1,
+}
+
+/// Methods for handling messages on the data port
+pub trait DataProcessor {
```

Comment:
I'm not happy with how this turned out and I plan to rewrite this in a follow-on PR

Reply:
Yeah, this is a little strange. What would we be blocking by rewriting now? Is there anything that blocks doing this better in this PR?